docker: fix the bug of using root account to create `/data/*` directory

eightpigs opened the pull request:
Fixes [#6067 ](https://github.com/gogs/gogs/issues/6067)

- Create /data/* directory with git user

unknwon commented on docker/start.sh:
```diff
@@ -43,6 +43,7 @@ setids() {
     PGID=${PGID:-1000}
     groupmod -o -g "$PGID" git
     usermod -o -u "$PUID" git
+    export USER=git
```
We can go one step further (not sure if it will work, looks like so):
```suggestion
    export USER=git
    groupmod -o -g "$PGID" $USER
    usermod -o -u "$PUID" $USER
```

unknwon commented:
Thanks for the PR! Left a comment.

eightpigs commented:
During my use, there are also permission errors in adding ssh key. I found that ~git/.ssh was created by root. 

So, I adjusted the *docker/s6/gogs/setup* file:
1. Create ~git/.ssh using git
2. Removed unnecessary permission settings

At present, everything works fine in the container.

unknwon commented:
Awesome, thanks!

unknwon commented:
Reverted because of https://github.com/gogs/gogs/commit/f3b05961aae82fa3e19c47f03c78974cf793fda3.
